src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/Context/Unwinder.cs

```diff
@@ -109,7 +109,15 @@ private static unsafe int ReadFromTarget(ulong address, void* pBuffer, int buffe
             return -1;
         }
         Span<byte> span = new Span<byte>(pBuffer, bufferSize);
-        callbackContext.Target.ReadBuffer(address, span);
+        try
+        {
+            callbackContext.Target.ReadBuffer(address, span);
+        }
+        catch (InvalidOperationException)
+        {
+            // if the read fails, the unwinder behavior changes. Return failing HR instead of throwing
```

Comment:
Can we specifically make sure we're only swallowing the type of IOEs we expect and not other IOEs from an actual bug in the implementation? Do we know what scenarios we expect a throw here in? It's fine if there's no way to sense what kind of IOE it is robustly (I don't expect you to parse .Message etc)

Reply:
OK, given the narrow scope of the code inside the try I'm fine with this for now, maybe just make a to-do note to clean it up later to not rely on exceptions.